### DISCUSSION TITLE
Support printing to stdout

### DISCUSSION THREAD

**Discussion by maxhawkins** [zoni/obsidian-export#61] [Feature requests]
It would be useful to support printing the output to stdout instead of providing a filename.

That way you could pipe the output to another program like:
```sh
obsidian-export doc.md | pandoc -f markdown -t html
```

**Comment by AB1908** [comment]
It's a "nice to have" for sure, but I can't see the necessity of it especially since we can `cat` the output. Could you explain a use-case where a pipe is a necessity?

I can't actually fix the issue or anything but I'm curious to see how you use it.

**Comment by maxhawkins** [comment]
It's a unix philosophy thing, mostly. Writing to stdout makes it easier to chain together programs without intermediate files.

Something like:

```
obsidian-export doc.md | sed 's/today/9-19-21/' | pandoc -f markdown -t html | ./upload-to-server.sh
```

It's especially important when you're processing a whole bunch at once (with parallel for instance) and don't want to litter your filesystem with temporary files you need to clean up later.

**Comment by AB1908** [comment]
Ah, fair enough. Thanks for the explanation.

**Comment by zoni** [comment]
I generally adhere strongly to Unix principles when designing CLI applications, but not supporting output to stdout has so far been a deliberate design decision here. Although I've since incorporated ways to more flexibly export individual files,`obsidian-export` was originally designed for bulk-export of entire vaults.

In many cases, when you invoke it, it will export multiple files. Exporting only a single file is a little bit of a special case right now.

If I were to support writing to stdout, what should happen when multiple files are being exported? Should it print everything to stdout, effectively concatenating all notes with no way to distinguish where one ends and another one begins? Should this cause an error?

(It's also likely that if I implement this, one will have to explicitly pass `-` as filename to write to stdout, and omitting `destination` won't automatically make it switch to writing to stdout.)
